src/Guesser/Guess/DateTimeType.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Joli\Jane\Guesser\Guess;
+
+use Joli\Jane\Generator\Context\Context;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Name;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Scalar;
+
+/**
+ * Represent a DateTime type.
+ */
+class DateTimeType extends ObjectType
+{
+    private $format;
+
+    public function __construct($object, $format = DATE_RFC3339)
+    {
+        parent::__construct($object, '\DateTime', []);
```

Comment:
Hmm thinking of just using DateTimeImmutable and nothing else, if someone wants a DateTime he can create it's own type, but don't to have too much generic thing, also supporting a Interface would be hard (it's change the way of generating type hint and other stuff)